platform-tests/src/test/java/org/junit/platform/launcher/TestIdentifierTests.java

```diff
@@ -57,34 +60,102 @@ void inheritsTypeFromDescriptor() {
 	}
 
 	@Test
-	void currentVersionCanBeSerializedAndDeserialized() throws Exception {
+	void currentVersionCanBeSerializedAndDeserialized() {
 		var originalIdentifier = createOriginalTestIdentifier();
-		var deserializedIdentifier = (TestIdentifier) deserialize(serialize(originalIdentifier));
-		assertDeepEquals(originalIdentifier, deserializedIdentifier);
+		var uniqueId = originalIdentifier.getUniqueIdObject();
+		var testSource = ClassSource.from(TestIdentifierTests.class);
+
+		var recreatedDescriptor = new AbstractTestDescriptor(uniqueId, originalIdentifier.getDisplayName(),
+			testSource) {
+			@Override
+			public Type getType() {
+				return Type.TEST;
+			}
+
+			@Override
+			public String getLegacyReportingName() {
+				return originalIdentifier.getLegacyReportingName();
+			}
+
+			@Override
+			public Set<TestTag> getTags() {
+				return Set.of(TestTag.create("aTag"));
+			}
+		};
+		var engineDescriptor = new EngineDescriptor(UniqueId.forEngine("engine"), "Engine");
+		engineDescriptor.addChild(recreatedDescriptor);
+
+		var rebuiltIdentifier = TestIdentifier.from(recreatedDescriptor);
+		assertDeepEquals(originalIdentifier, rebuiltIdentifier);
 	}
 
 	@Test
-	void initialVersionCanBeDeserialized() throws Exception {
-		try (var inputStream = getClass().getResourceAsStream("/serialized-test-identifier")) {
```

Comment:
I think we can remove this test and the corresponding file (`platform-tests/src/test/resources/serialized-test-identifier`).